pkgs/native_toolchain_c/test/cbuilder/cbuilder_test.dart

```diff
@@ -200,6 +200,222 @@ void main() {
       () => testDefines(customDefineWithValue: value),
     );
   }
+
+  test('CBuilder flags', () async {
+    await inTempDir(
+      // https://github.com/dart-lang/sdk/issues/40159
+      keepTemp: Platform.isWindows,
+      (tempUri) async {
+        final addCUri =
+            packageUri.resolve('test/cbuilder/testfiles/add/src/add.c');
+        const name = 'add';
+
+        final logMessages = <String>[];
+        final logger = createCapturingLogger(logMessages);
+
+        final buildConfig = BuildConfig(
+          outDir: tempUri,
+          packageRoot: tempUri,
+          targetArchitecture: Architecture.current,
+          targetOs: OS.current,
+          buildMode: BuildMode.release,
+          linkModePreference: LinkModePreference.dynamic,
+          cCompiler: CCompilerConfig(
+            cc: cc,
+            envScript: envScript,
+            envScriptArgs: envScriptArgs,
+          ),
+        );
+        final buildOutput = BuildOutput();
+
+        final flag = switch (buildConfig.targetOs) {
+          OS.windows => '/O2',
```

Comment:
Can we compile with some flags we can verify that they were passed?

For example `-rdynamic` and checking for a symbol in an executable?
Or with different optimization levels and comparing the sizes?

Reply:
I've reused the `defines.c` test file, but with passing the define flag directly through the `flags` option.